self-healing/heal-ldes-data.ts

```diff
@@ -54,13 +59,41 @@ async function triggerRecreate(differences) {
   ]);
 }
 
-async function getSubjectTypes(subjects: string[]) {
+async function getSubjectTypes(subjects: string[], stream, config) {
+  const graphTypesToExclude = config[stream].graphTypesToExclude;
+  const excludedGraphs = config[stream].graphsToExclude;
+  let excludeGraphsFilter = "";
+  if (excludedGraphs?.length > 0) {
+    excludeGraphsFilter = excludedGraphs
+      .map((graph: string) => sparqlEscapeUri(graph))
+      .join(", ");
+    excludeGraphsFilter = `FILTER(?targetGraph NOT IN (${excludeGraphsFilter}))`;
+  }
+  let excludeGraphTypesFilter = "";
+  let excludeGraphTypeValues = "";
+  if (graphTypesToExclude?.length > 0) {
+    excludeGraphTypeValues = graphTypesToExclude
+      .map((type: string) => sparqlEscapeUri(type))
+      .join("\n ");
+    excludeGraphTypeValues = `VALUES ?excludeGraphType { ${excludeGraphTypeValues} }`;
+    excludeGraphTypesFilter = `FILTER NOT EXISTS {
+      ?targetGraph a ?excludedGraphType.
+    }`;
+  }
```

Comment:
Shouldn't this be extracted, since this exact same piece of code is used twice?

Reply:
yes, but it triggered me to create a ticket to generalize this and given that that ticket now exists and this will all be rewritten, i think we can keep it for now